packages/@azure/eventhubs/client/lib/connectionContext.ts

```diff
@@ -116,6 +116,15 @@ export namespace ConnectionContext {
         numSenders: Object.keys(connectionContext.senders).length,
         numReceivers: Object.keys(connectionContext.receivers).length
       };
+
+      // Remove all the links and sessions for this AMQP connection.
+       connectionContext.connection.removeAllSessions();
+
+         // Close the cbs session
+         await connectionContext.cbsSession.close();
+         // Close the management session
+         await connectionContext.managementSession!.close();
```

Comment:
I am afraid no one is recreating the management session :(
cbs session will get created when any of the sender/receiver is detached

Reply:
Never mind, every feature of the managementClient internally calls _init() which ensures to create the management link if it is not open already

Reply:
But let's run a test to ensure this regardless